src/lib/block_if.c

```diff
@@ -618,17 +618,17 @@ blockif_open(const char *optstr, const char *ident)
 	sectsz = DEV_BSIZE;
 	psectsz = psectoff = 0;
 	candelete = geom = 0;
+  blocks = 0;
 	if (S_ISCHR(sbuf.st_mode)) {
-		perror("xhyve: raw device support unimplemented");
-		goto err;
-		// if (ioctl(fd, DIOCGMEDIASIZE, &size) < 0 ||
-		// 	ioctl(fd, DIOCGSECTORSIZE, &sectsz))
-		// {
-		// 	perror("Could not fetch dev blk/sector size");
-		// 	goto err;
-		// }
-		// assert(size != 0);
-		// assert(sectsz != 0);
+		if (ioctl(fd, DKIOCGETBLOCKCOUNT, &blocks) < 0 ||
+        ioctl(fd, DKIOCGETBLOCKSIZE, &sectsz)) {
+		  perror("Could not fetch dev blk/sector size");
+		 	goto err;
+		}
+		assert(blocks != 0);
+		assert(sectsz != 0);
```

Comment:
These should probably be dynamic checks that also `perror`, in case we compile with asserts disabled.